### DISCUSSION TITLE
Deleting isolated parts of a mesh

### DISCUSSION THREAD

**Discussion by drathin4** [libigl/libigl#1916] [Q&A]
I want to remove the isolated regions of an STL mesh as a post-processing step in an operation. These unfortunately contribute to the "noise". I tried playing around with APIs like `connected_components, facet_components, adjacency_matrix` etc., but haven't been able to figure it out. Could you please help ?

**Comment by drathin4** [comment]
@jdumas @alecjacobson

**Comment by alecjacobson** [comment]
Could you describe your desired input and output? even better to include a
mesh or picture

**Reply by drathin4** [reply]
I'm sending 2 attachments. 1 containing the image and other containing the .stl file. The image contains the isolated region highlighted in **RED** color which should not be present in the output.

![Capture](https://user-images.githubusercontent.com/79855313/141478422-c6ea4127-2741-4b5f-be32-1aac7336beaf.PNG)

This is the mesh file.
[libIGL_Result.zip](https://github.com/libigl/libigl/files/7528018/libIGL_Result.zip)

**Comment by alecjacobson** [comment]
connected_components is the right thing to use. A common mistake when
working with .stl files is to forget to merge vertices after loading (an
.stl file stores an unconnected list of triangles, not an indexed mesh).
You can do that with remove_duplicate_vertices. For each component you
could then compute the surface area (sum of igl::double_area) or volumes
(second output in igl::centroid). Then you could filter small components or
keep just the biggest one.

**Reply by drathin4** [reply]
Thanks for the reply. As you guessed correctly, I forgot to merge the vertices of STL file, and hence it was not giving the desired results. Otherwise my algorithm was correct.
